tests/ut/language/parser/test_subscript_syntax.py

```
@@ -729,6 +799,41 @@ def write_row(
+            t[0:16, 0:4] = narrowed
+            return pl.store(t, [0, 0], x)
+
+        printed = narrow_tile_write.as_python()
+        assert "tile.assemble" in printed
+
+    def test_tile_subscript_write_rank_reduce_narrow_valid_rejected(self):
+        """Tile rank-reduce + narrow valid_shape is rejected: tile.reshape
+        cannot carry valid_shape, so the rank lift would silently lose the
+        narrowing. Force users to take the pl.store path instead."""
+
+        with pytest.raises(UnsupportedFeatureError, match="tile.reshape cannot carry valid_shape"):
```

Ruff (0.15.14)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)